### PR TITLE
Add ownerhover script

### PR DESCRIPTION
**Pull request by szero**
As requested by Liquid

### CODE REVIEW

**Comment by szero** [comment]
now with janitors support

File: titlehover.user.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+// ==UserScript==
+// @name         StuffOnNameHover
+// @namespace    https://not.jew.dance
+// @version      3
+// @description  Show da owner
+// @author       Your mom
+// @icon         https://volafile.org/favicon.ico
+// @match        https://volafile.org/r/*
+// @require      https://cdn.jsdelivr.net/gh/volafiled/volascripts/dry.js
```
Comment (RealDolos) [inline titlehover.user.js]: require specific commit of dry.js, not master.

File: titlehover.user.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+// @grant        none
+// @run-at       document-start
+// ==/UserScript==
+/* globals dry */
+
+dry.once("load", () => {
+  'use strict';
+  String.prototype.capitalize = function() {
+    return this[0].toUpperCase() + this.slice(1).toLowerCase();
```
Comment (RealDolos) [inline titlehover.user.js]: No prototype expansion of built-in objects pls.

File: titlehover.user.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  };
+  const room_title = document.getElementById("name_container");
+  let room_owner;
+  let jannies = new Array();
```
Comment (RealDolos) [inline titlehover.user.js]: []

File: titlehover.user.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  dry.exts.connection.once("config", cfg => {
```
Comment (RealDolos) [inline titlehover.user.js]: shouldn't this calculate janitors too?

**Review by RealDolos — CHANGES_REQUESTED** [review]
The entire logic is duplicated a bunch of times (and incorrect as sometimes it will only consider the room owner).
Just have some variables tracking owner and janitors, and one function to actually set/update .title according to that.
I also do not like that you capitalize names, tbh. this is imo less correct than just keeping the lower-case stuff. If you really wanted to look up the current casing of a name, there would be ways, but that's trading a lot of overhead for limited utility.

File: titlehover.user.js (hunk 4 above)
Comment (szero) [reply titlehover.user.js]: its the only listener naow

**Review by RealDolos — CHANGES_REQUESTED** [review]
Also, titlehover is  a shit file name, and the name and description aren't great either.
Also it's not "jannies"".

You only ever do stuff onload. Is that actually early enough to get the initial config always? IIRC the connect setup starts ondom so before onload and could in theory finish before onload.

**Comment by szero** [comment]
I can't use extensions when I want to do stuff on dom load because they're not available yet, I could set some setInterval thing with low timeout that waits until exts is not null, but the on load listener works reliably enough.

**Comment by szero** [comment]
If it's not "jannies" than what is it? What is plural form of "janny"?

**Comment by RealDolos** [comment]
As already pointed out, the logic for per-user select is broken, looking at `e.target` indiscriminately, therefore breaking when the wrong element within a file element is clicked.

**Pull request merged**